### PR TITLE
Use backports.ssl_match_hostname

### PR DESCRIPTION
**Pull request by felixonmars**
The py2-ipaddress module unfortunately conflicts with the pypi:ipaddress module, which is in the dependency tree of widely used pyOpenSSL. I think it would be a good idea to use a well maintained backport of the Python 3.5 implementation of match_hostname() instead of duplicating the effort and maintain another.

All tests are passing here.

Signed-off-by: Felix Yan <felixonmars@archlinux.org>

### CODE REVIEW

**Comment by felixonmars** [comment]
Hmm, this is really weird. The IP Address test passes here with Python 2.7.11, and also passes on the Jenkins build, but failed on Travis.

**Comment by felixonmars** [comment]
Oops, nevermind. Turns out the `ipaddress` module is also needed. Updated and all should work fine now!

**Comment by felixonmars** [comment]
Looks like the Jenkins failure is not related to this PR.

File: setup.py [hunk 1]
```diff
@@ -13,7 +13,8 @@
 ]
 
 extras_require = {
-    ':python_version < "3"': 'py2-ipaddress >= 3.4.1',
+    ':python_version < "3.5"': 'backports.ssl_match_hostname >= 3.5',
+    ':python_version < "3.5"': 'ipaddress',
```
Comment (shin-) [inline setup.py]: I think this is only needed for `python < 3.3` ( https://docs.python.org/3/library/ipaddress.html )

**Comment by shin-** [comment]
Thanks! I didn't know this had been backported already. Obviously it is much better than having to maintain it ourselves.

Just one small issue with the requirements, otherwise this is good.

File: setup.py (hunk 1 above)
Comment (felixonmars) [inline setup.py]: According to https://github.com/felixonmars/docker-py/blob/backports/docker/ssladapter/ssladapter.py#L18 and https://bugs.python.org/issue23239, the IP address support wasn't backported to Python 3.3 and 3.4 (yet?), so when I use `< 3.3` the tests failed.

**Comment by TomasTomecek** [comment]
This is awesome! I'm currently struggling with docker-py 1.8 in Fedora world since we don't have `py2-ipaddress` packaged.

File: setup.py (hunk 1 above)
Comment (shin-) [inline setup.py]: `ipaddress` is available from python 3.3 onwards. Change in #1039 is what I meant and seems to be working fine.

Comment (felixonmars) [inline setup.py]: Ah, I see. Thanks :)

**Pull request merged**